test/core_ext/deep_prune_test.rb

```diff
@@ -0,0 +1,54 @@
+# encoding: utf-8
+
+require File.expand_path(File.join(File.dirname(__FILE__) + '/../test_helper'))
+require File.expand_path(File.join(File.dirname(__FILE__) + '/../../lib/core_ext/hash/deep_prune'))
+
+class TestDeepprune < Test::Unit::TestCase
+  test "deep_prune! returns empty Hash for empty Hash" do
+    assert_equal({}, {}.deep_prune!)
+  end
+
+  test "deep_prune! with default comparator calls deep_prune! recursively and removes empty hashes" do
+    hash = {
+      "nil_value" => nil,
+      "empty_hash" => {},
+      "nested_empty_hash" => {
+        "foo" => {},
+      },
+      "empty_string" => "",
+      "empty_array" => [],
+      "integer" => 1,
+    }
```

Comment:
Nit: We could include sub-hashes that will stay in the expected result, just to make sure all hashes are not just deleted by `deep_prune`

Reply:
Added. 👍